### PR TITLE
Feat custom tags

### PR DESCRIPTION
**Pull request by samuel404**
added custom tags to boards and cards

### CODE REVIEW

**Comment by ShragaUser** [comment]
commits are bad - not explanatory at all!

File: src/app/components/Board/Board.jsx [hunk 1]
```diff
@@ -140,8 +142,77 @@ class Board extends Component {
...
   };
+  checkFormat= (lists) => {
+    const { dispatch} = this.props;
+    const oldLabels = ["inprogress","general","tracking","bug","help","critical",]
+    let cardsId=[];
+    let newLabelFormat=[];
+    lists.forEach(list => {
+      cardsId = cardsId.concat(list.cards);
+    });
+    console.log(this.props.boardLabels);
```
Comment (ShragaUser) [inline src/app/components/Board/Board.jsx]: delete console.log()

File: src/app/components/Board/Board.jsx [hunk 2]
```diff
@@ -140,8 +142,77 @@ class Board extends Component {
...
+    cardsId.forEach(cardId => {
+      if(this.props.cards[cardId].labels && oldLabels.includes(this.props.cards[cardId].labels[0])){
```
Comment (ShragaUser) [inline src/app/components/Board/Board.jsx]: this.props.cards should be const { cards } = this.props;

File: src/app/components/Board/Board.jsx [hunk 3]
```diff
@@ -140,8 +142,77 @@ class Board extends Component {
       window.scrollTo(window.scrollX - 80, 0);
     }
   };
+  checkFormat= (lists) => {
+    const { dispatch} = this.props;
+    const oldLabels = ["inprogress","general","tracking","bug","help","critical",]
+    let cardsId=[];
+    let newLabelFormat=[];
+    lists.forEach(list => {
+      cardsId = cardsId.concat(list.cards);
```
Comment (ShragaUser) [inline src/app/components/Board/Board.jsx]: no need. 
just go over each card in list.  and do whatever is needed.

File: src/app/components/Board/Board.jsx [hunk 4]
```diff
@@ -140,8 +142,77 @@ class Board extends Component {
...
+    });
+    console.log(this.props.boardLabels);
+    cardsId.forEach(cardId => {
```
Comment (ShragaUser) [inline src/app/components/Board/Board.jsx]: take out to different function.

File: src/app/components/Board/Board.jsx [hunk 5]
```diff
@@ -140,8 +142,77 @@ class Board extends Component {
...
+      if(this.props.cards[cardId].labels && oldLabels.includes(this.props.cards[cardId].labels[0])){
+        newLabelFormat=[];
+        this.props.cards[cardId].labels.forEach(label => {
+          switch (label) {
+            case "inprogress":
+              this.props.boardLabels.forEach(label => {
```
Comment (ShragaUser) [inline src/app/components/Board/Board.jsx]: deconstruct boardLabels

File: src/app/components/Board/Board.jsx [hunk 6]
```diff
@@ -140,8 +142,77 @@ class Board extends Component {
...
+              break;
+          }
+        });
+        dispatch({
+          type: "FIX_LABELS_FORMAT",
+          payload: { newLabelFormat, cardId }
+        });
+      }
+    });
+  }
```
Comment (ShragaUser) [inline src/app/components/Board/Board.jsx:174]: this whole function should be refactored. 
can be a lot more efficient!

File: src/app/components/BoardHeader/ColorPicker.jsx [hunk 7]
```diff
@@ -15,6 +15,7 @@ class ColorPicker extends Component {
     boardId: PropTypes.string.isRequired,
     boardColor: PropTypes.string.isRequired,
     dispatch: PropTypes.func.isRequired
+
```
Comment (ShragaUser) [inline src/app/components/BoardHeader/ColorPicker.jsx]: delete space

File: src/app/components/BoardList/BoardList.jsx [hunk 8]
```diff
@@ -75,10 +99,12 @@ class BoardList extends Component {
     )};
 }
 
-const mapStateToProps = ({ listsById,user, socketConnected }) => ({
+const mapStateToProps = ({ listsById,user, socketConnected }) => {
+  return{
```
Comment (ShragaUser) [inline src/app/components/BoardList/BoardList.jsx]: no need for return. 
can be implicit

File: src/app/components/CardBadges/CardBadges.jsx [hunk 9]
```diff
@@ -21,8 +21,30 @@ class CardBadges extends Component {
...
+    labels: PropTypes.array,
+    boardId: PropTypes.string.isRequired,
+    cardId: PropTypes.string.isRequired,
+    dispatch: PropTypes.func.isRequired
   };
+  constructor() {
+    super();
+  }
+/*
+  shouldComponentUpdate(nextProps){
```
Comment (ShragaUser) [inline src/app/components/CardBadges/CardBadges.jsx]: delete if not necessary

File: src/app/components/CardBadges/CardBadges.jsx [hunk 10]
```diff
@@ -114,35 +136,57 @@ class CardBadges extends Component {
...
+    }
+  };
 
-  renderLabels = () => {
-    const { labels, t } = this.props;
-    const colorsWithLabelsMap = new Map(colorsWithLabels);
+  getLabelById = (labelId) => {
+    const { dispatch, cardId } = this.props;
+    let labelvalue;
+    this.props.boardLabels.map(label => {
```
Comment (ShragaUser) [inline src/app/components/CardBadges/CardBadges.jsx:140]: deconstruct

File: src/app/components/CardBadges/CardBadges.jsx [hunk 11]
```diff
@@ -162,4 +206,11 @@ class CardBadges extends Component {
   }
 }
 
-export default connect()(withTranslation()(CardBadges));
+const mapStateToProps = (state, ownProps) => {
+  const boardLabel = state.boardsById[ownProps.boardId].labels;
```
Comment (ShragaUser) [inline src/app/components/CardBadges/CardBadges.jsx]: just rename to boardLabels and then return { boardLabels }

File: src/app/components/CardModal/CardOptions.jsx [hunk 12]
```diff
@@ -23,18 +25,27 @@ class CardOptions extends Component {
...
     toggleColorPicker: PropTypes.func.isRequired,
-    dispatch: PropTypes.func.isRequired
+    dispatch: PropTypes.func.isRequired,
+    boardId: PropTypes.string.isRequired,
   };
 
   constructor() {
     super();
+
+    this.toggelEdit = this.toggelEdit.bind(this);
```
Comment (ShragaUser) [inline src/app/components/CardModal/CardOptions.jsx]: why though?

File: src/app/components/CardModal/CardOptions.jsx [hunk 13]
```diff
@@ -45,19 +56,46 @@ class CardOptions extends Component {
...
       dispatch({
         type: "ADD_LABEL",
-        payload: { label, cardId: card._id }
+        payload: { label: label.id, cardId: card._id }
       });
     }
+
+  };
+
+  editModeCheack = label => {
```
Comment (ShragaUser) [inline src/app/components/CardModal/CardOptions.jsx]: check

**Review by ShragaUser — CHANGES_REQUESTED** [review]
remove all comments 
remove all console.log() 
always use deconstruct
refactor functions -> too long and too much responsibility.

File: src/app/components/Board/Board.jsx [hunk 14]
```diff
@@ -142,72 +142,74 @@ class Board extends Component {
...
     }
   };
+
+  updateFormat = (cardId,boardLabels) => {
+    const {dispatch,cards} = this.props;
+    let newLabelFormat=[];
+    cards[cardId].labels.forEach(label => {
+      switch (label) {
+        case "inprogress":
+          boardLabels.forEach(label => {
```
Comment (ShragaUser) [inline src/app/components/Board/Board.jsx]: create single map object with key being: "inprogress" and value veing id. 
then just use labelIdMap[label]...

**Review by ShragaUser — CHANGES_REQUESTED** [review]
simple change

**Review by dorweitman — APPROVED** [review]
Shit work

**Review by ShragaUser — APPROVED** [review]
Yas

**Pull request merged**